docs/source/howto/codes/interfacing.rst

```diff
@@ -0,0 +1,282 @@
+We will see cases for both.
+
+.. code:: python
+
+        @classmethod
+        def define(cls, spec):
+            super().define(spec)
+
+            spec.output('sum', valid_type=orm.Int, help='The sum of the left and right operand.')
+            spec.input('x', valid_type=orm.Int, help='The left operand.')
+            spec.input('y', valid_type=orm.Int, help='The right operand.')
+
+            spec.inputs['metadata']['options']['parser_name'].default = 'arithmetic.add'
+            spec.inputs['metadata']['options']['input_filename'].default = 'aiida.in'
+            spec.inputs['metadata']['options']['output_filename'].default = 'aiida.out'
+
+            spec.exit_code(300, 'ERROR_NO_RETRIEVED_FOLDER', message='The retrieved folder data node could not be accessed.')
+            spec.exit_code(310, 'ERROR_READING_OUTPUT_FILE', message='The output file could not be read from the retrieved folder.')
+            spec.exit_code(320, 'ERROR_INVALID_OUTPUT', message='The output file contains invalid output.')
+
+The first line just runs the ``define`` method of the parent base |CalcJob| class, which will define some basic `inputs` and `outputs` (as we will see shortly).
+The second block, however, starts defining some of the `inputs` and `outputs` that are specific to the use of this code.
+One can see that it does so in the same way for both: by providing an identifier (or label: ``'x'``, ``'y'`` or ``'sum'``), a valid type (in this case AiiDA nodes of type |Int|) and a help message for the users of the class.
+
+The next block seems similar but it is actually different: it is not defining new inputs but modifying some properties for inputs that are already defined in the base |CalcJob| class.
+You can spot the difference in that defining is performed by using the ``spec.input`` method, singular, whereas accessing is achieved via the ``spec.inputs`` method, plural.
+This change here is merely for convenience to provide `default` values for some relevant variables that will be needed for the next methods so that the user doesn't have to provide them when instantiating this |CalcJob| to run the code.
+Note that these are not input nodes but options you set directly on the |CalcJob|, yet they are defined pretty much in the same way:
+
+.. code:: python
+
+        spec.input(
+            'metadata.options.input_filename',
+            valid_type=str,
+            required=False,
+            help='Filename to which the input for the code that is to be run is written.'
+        )
+
+You can see that this is input is defined with ``required=False``, so the user can ``run`` the code without specifying it in its instantiated |CalcJob|.
+This is not the case for the specific inputs we defined (since ``required=True`` by default), so if the user tries to ``run`` or ``submit`` the |CalcJob| without providing these, the code will raise.
```

Comment:
```suggestion
This is not the case for the specific inputs we defined (since ``required=True`` by default), so if the user tries to ``run`` or ``submit`` the |CalcJob| without providing these, the code will raise an exception.
```